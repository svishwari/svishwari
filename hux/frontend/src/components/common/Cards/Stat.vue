<template>
  <v-menu
    bottom
    offset-y
    offset-x
    nudge-left="40px"
    nudge-top="10px"
    open-on-hover
  >
    <template #activator="{ on }">
      <div class="px-4 border-start border-zircon" :class="statClass" v-on="on">
        <h4 class="text-h4 mb-0">{{ value }}</h4>
        <div class="text-caption black--text text--darken-4">{{ label }}</div>
      </div>
    </template>
    <div v-if="$slots.default" class="px-4 py-3 text-caption white">
      <slot name="default" />
    </div>
  </v-menu>
</template>

<script>
export default {
  name: "Stat",

  props: {
    label: {
      type: String,
      required: false,
    },
    value: {
      type: String,
      required: false,
    },
    statClass: {
      type: String,
      required: false,
    },
  },
}
</script>
