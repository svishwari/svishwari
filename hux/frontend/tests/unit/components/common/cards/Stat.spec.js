import Stat from "@/components/common/Cards/Stat"
import { mount } from "@vue/test-utils"

describe("Cards", () => {
  describe("Card Stat", () => {
    test("Displays with custom properties", () => {
      const customProps = {
        label: "Last trained",
        value: "2 hrs ago",
      }

      const wrapper = mount(Stat, {
        propsData: customProps,
        slots: {
          default: "12:45pm",
        },
      })

      expect(wrapper.text()).toContain(customProps.label)
      expect(wrapper.text()).toContain(customProps.value)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
