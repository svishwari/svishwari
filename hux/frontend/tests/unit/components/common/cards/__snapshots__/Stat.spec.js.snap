// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards Card Stat Displays with custom properties 1`] = `
<div class="v-menu">
  <div class="px-4 border-start separator">
    <div class="text-h4 black--text">2 hrs ago</div>
    <div class="black--text text--lighten-4">Last trained</div>
  </div>
  <!---->
</div>
`;
